tests/components/whirlpool/test_climate.py

```diff
@@ -0,0 +1,343 @@
+        [mock.call(False), mock.call(True)]
+    )
+
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: "climate.said1"},
+        blocking=True,
+    )
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_HVAC_MODE: HVAC_MODE_COOL},
+        blocking=True,
+    )
+    mock_aircon_api.return_value.set_power_on.assert_has_calls(
+        [mock.call(False), mock.call(True)]
+    )
+
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_TEMPERATURE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_TEMPERATURE: 15},
+        blocking=True,
+    )
+    mock_aircon_api.return_value.set_temp.assert_called_once_with(15)
+
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_HVAC_MODE: HVAC_MODE_COOL},
+        blocking=True,
+    )
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_HVAC_MODE: HVAC_MODE_HEAT},
+        blocking=True,
+    )
+    # HVAC_MODE_DRY should be ignored
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_HVAC_MODE: HVAC_MODE_DRY},
+        blocking=True,
+    )
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_HVAC_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_HVAC_MODE: HVAC_MODE_FAN_ONLY},
+        blocking=True,
+    )
+    mock_aircon_api.return_value.set_mode.assert_has_calls(
+        [
+            mock.call(whirlpool.aircon.Mode.Cool),
+            mock.call(whirlpool.aircon.Mode.Heat),
+            mock.call(whirlpool.aircon.Mode.Fan),
+        ]
+    )
+
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_FAN_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_FAN_MODE: FAN_AUTO},
+        blocking=True,
+    )
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_FAN_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_FAN_MODE: FAN_LOW},
+        blocking=True,
+    )
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_FAN_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_FAN_MODE: FAN_MEDIUM},
+        blocking=True,
+    )
+    # FAN_MIDDLE should be ignored
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_FAN_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_FAN_MODE: FAN_MIDDLE},
+        blocking=True,
+    )
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_FAN_MODE,
+        {ATTR_ENTITY_ID: "climate.said1", ATTR_FAN_MODE: FAN_HIGH},
+        blocking=True,
+    )
+
+    mock_aircon_api.return_value.set_fanspeed.assert_has_calls(
```

Comment:
Assert after each call to ensure each call works as expected instead of the end result.